### PR TITLE
Add bitfields to metadata

### LINKED ISSUE #17: Can we extract the bit masks used for entity metadata parsing
Like here: https://wiki.vg/Entity_metadata#Entity to extract this table 

bit mask | description
-- | --
0x01 | Is on fire
0x02 | Is crouching
0x04 | Unused (previously riding)
0x08 | Is sprinting
0x10 | Is swimming
0x20 | Is invisible
0x40 | has glowing effect
0x80 | Is flying with an elytra

and here: https://wiki.vg/Entity_metadata#Living_Entity

bit mask | description
-- | --
0x01 | Is hand active
0x02 | Active hand (0 = main hand, 1 = offhand)
0x04 | Is in riptide spin attack

and so on

### PR DESCRIPTION
**Pull request by mat-1**
Adds a "bitfields" parameter in the metadata which includes method, value (bitmask), and (sometimes) class
screenshot of abstract_entity bitfields

Closes #17 

(i hope i rebased correctly this time lol)

### CODE REVIEW

**Comment by Pokechu22** [comment]
Code looks good to me and seems to run correctly on all versions where entity metadata works (I tested 1.9, 1.12.2, and 21w07a).  I think `mask` might be a clearer name than `value` though.  There are also a few minor spelling and whitespace errors, but I can fix all of those up afterwards.

The rebase doesn't seem to have gone perfectly, but I can do a squash merge which should avoid causing issues in the main repo (it'll also only create one commit, so the extra fixup commits won't show up).

**Comment by mat-1** [comment]
> 
> 
> Code looks good to me and seems to run correctly on all versions where entity metadata works (I tested 1.9, 1.12.2, and 21w07a). I think `mask` might be a clearer name than `value` though. There are also a few minor spelling and whitespace errors, but I can fix all of those up afterwards.
> 
> The rebase doesn't seem to have gone perfectly, but I can do a squash merge which should avoid causing issues in the main repo (it'll also only create one commit, so the extra fixup commits won't show up).

Ok, I changed it from `value` to `mask` and fixed some of those issues

File: burger/toppings/entitymetadata.py [hunk 1]
```diff
@@ -231,8 +236,76 @@ def on_new(self, ins, const):
...
+                        # the method calls getField() or getSharedField()
+                        if ins.mnemonic in ("invokevirtual", "invokespecial", "invokeinterface", "invokestatic"):
+                            calling_method = ins.operands[0].name_and_type.name.value
+
+                            has_correct_arguments = ins.operands[0].name_and_type.descriptor.value == "(I)Z"
+
+                            is_getflag_method = has_correct_arguments and calling_method == get_flag_method
+                            is_shared_getflag_method = has_correct_arguments and calling_method == shared_get_flag_method
+
+                            # if its a shared flag, update the bitfields_by_class for abstract_entity
```
Comment (Pokechu22) [inline burger/toppings/entitymetadata.py]: Since you seem to have addressed everything else, this should be <q>it's</q>.

**Comment by Pokechu22** [comment]
Merged, thanks!  This data will be included in the next snapshot burger runs on (probably 21w08a).  I'll eventually re-run burger on _all_ the snapshots, but that won't happen for a while.

I did just now think of one other thing that might be useful: getting the actual metadata field associated with the bitfield.  But currently, no entity has more than one bitfield, so I don't think that's too important; if that ever changes I don't think it'd be too hard to add.

**Comment by mat-1** [comment]
I'm like 90% sure the first byte is always the bitfield

**Comment by Pokechu22** [comment]
It's the last byte for [pandas](https://wiki.vg/Entity_metadata#Panda).

**Pull request merged**